boa_engine/src/vm/opcode/generator/mod.rs

```diff
@@ -1,20 +1,172 @@
 pub(crate) mod yield_stm;
 
+use std::collections::VecDeque;
+
 use crate::{
+    builtins::{
+        async_generator::{AsyncGenerator, AsyncGeneratorState},
+        generator::{GeneratorContext, GeneratorState},
+    },
+    environments::EnvironmentStack,
     error::JsNativeError,
+    object::{ObjectData, PROTOTYPE},
     string::utf16,
     vm::{
         call_frame::GeneratorResumeKind,
         opcode::{Operation, ReThrow},
-        CompletionType,
+        CallFrame, CompletionType,
     },
-    Context, JsError, JsResult,
+    Context, JsError, JsObject, JsResult,
 };
 
 pub(crate) use yield_stm::*;
 
 use super::SetReturnValue;
 
+/// `Generator` implements the Opcode Operation for `Opcode::Generator`
+///
+/// Operation:
+///  - Creates the generator generator object and yields.
```

Comment:
Is the double generator intended?

```suggestion
///  - Creates the generator object and yields.
```